File: fix_all_now.py
#!/usr/bin/env python3
"""
🔧 Correction automatique COMPLÈTE des imports lucide-react et sonner
Script pour SmartCabb - Correction de TOUS les fichiers
"""

import os
import re
from pathlib import Path

def fix_file(filepath):
    """Corrige les imports lucide-react et sonner dans un fichier"""
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Correction pattern 1: from 'lucide-react';
        content = re.sub(
            r"from\s+['\"]lucide-react['\"];",
            "from 'lucide-react@0.550.0';",
            content
        )
        
        # Correction pattern 2: from 'lucide-react' (sans point-virgule, pour les multilignes)
        content = re.sub(
            r"from\s+['\"]lucide-react['\"](?!@)",
            "from 'lucide-react@0.550.0'",
            content
        )
        
        # Correction pattern 3: from 'sonner';
        content = re.sub(
            r"from\s+['\"]sonner['\"];",
            "from 'sonner@2.0.3';",
            content
        )
        
        # Correction pattern 4: from 'sonner' (sans point-virgule)
        content = re.sub(
            r"from\s+['\"]sonner['\"](?!@)",
            "from 'sonner@2.0.3'",
            content
        )
        
        # Si le fichier a été modifié, le sauvegarder
        if content != original_content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
            return True
        
        return False
        
    except Exception as e:
        print(f"❌ Erreur dans {filepath}: {e}")
        return False

def main():
    print("╔════════════════════════════════════════════════════════════════╗")
    print("║  🔧 CORRECTION AUTOMATIQUE - SMARTCABB                         ║")
    print("║  Correction de TOUS les imports lucide-react et sonner        ║")
    print("╚════════════════════════════════════════════════════════════════╝")
    print()
    
    # Trouver tous les fichiers TypeScript et TSX
    all_files = []
    for pattern in ['**/*.ts', '**/*.tsx']:
        all_files.extend(Path('.').glob(pattern))
    
    # Filtrer les fichiers à ignorer
    files_to_process = [
        f for f in all_files
        if 'node_modules' not in str(f)
        and '.git' not in str(f)
        and 'dist' not in str(f)
        and 'build' not in str(f)
        and '.next' not in str(f)
    ]
    
    print(f"📂 Fichiers trouvés : {len(files_to_process)}\n")
    
    fixed_count = 0
    fixed_files = []
    
    for filepath in files_to_process:
        if fix_file(filepath):
            fixed_count += 1
            fixed_files.append(str(filepath))
            print(f"✅ {filepath}")
    
    print()
    print("╔════════════════════════════════════════════════════════════════╗")
    print(f"║  ✅ TERMINÉ ! {fixed_count} fichier(s) corrigé(s)")
    print("╚════════════════════════════════════════════════════════════════╝")
    print()
    
    if fixed_count > 0:
        print("📋 Résumé des corrections :")
        print("   • from 'lucide-react' → from 'lucide-react@0.550.0'")
        print("   • from 'sonner' → from 'sonner@2.0.3'")
        print()
        
        if len(fixed_files) <= 30:
            print(f"📁 Fichiers modifiés ({len(fixed_files)}) :")
            for f in fixed_files:
                print(f"   • {f}")
        else:
            print(f"📁 Fichiers modifiés ({len(fixed_files)}) :")
            for f in fixed_files[:20]:
                print(f"   • {f}")
            print(f"   ... et {len(fixed_files) - 20} autres fichiers")
    
    print()
    print("🚀 Prochaines étapes :")
    print("   1. Les fichiers sont maintenant corrigés")
    print("   2. Copiez-les vers votre dépôt GitHub")
    print("   3. Commit + Push")
    print("   4. Vercel déploiera automatiquement")
    print()

if __name__ == '__main__':
    main()

File: fix-all-now.js
#!/usr/bin/env node
/**
 * 🚀 Script de correction ULTRA-RAPIDE des imports
 * Corrige TOUS les fichiers en une seule passe
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

console.log('\n🚀 SmartCabb ULTRA-FIX v517.124\n');
console.log('='.repeat(60));

const files = [
  // Driver
  '/components/driver/DriverDashboard.tsx',
  '/components/driver/DriverLoginScreenNew.tsx',
  '/components/driver/DriverProfileScreen.tsx',
  '/components/driver/DriverRegistrationScreen.tsx',
  '/components/driver/DriverSettingsScreen.tsx',
  '/components/driver/DriverWalletScreen.tsx',
  '/components/driver/DriverWelcomeScreen.tsx',
  '/components/driver/EarningsScreen.tsx',
  '/components/driver/NavigationScreen.tsx',
  '/components/driver/NewRideNotification.tsx',
  '/components/driver/ConfirmationCodeScreen.tsx',
  '/components/driver/ActiveRideScreen.tsx',
  '/components/driver/PaymentConfirmationScreen.tsx',
  
  // Passenger
  '/components/passenger/AlternativeVehicleDialog.tsx',
  '/components/passenger/BookForSomeoneElse.tsx',
  '/components/passenger/CancelRideReasonModal.tsx',
  '/components/passenger/EstimateScreen.tsx',
  '/components/passenger/FavoriteLocations.tsx',
  '/components/passenger/PaymentMethodScreen.tsx',
  '/components/passenger/PaymentReceiptScreen.tsx',
  '/components/passenger/PaymentScreen.tsx',
  '/components/passenger/PrivacySettingsScreen.tsx',
  '/components/passenger/ProfileScreen.tsx',
  '/components/passenger/PromoCodeScreen.tsx',
  '/components/passenger/RatingDialog.tsx',
  '/components/passenger/RatingModal.tsx',
  '/components/passenger/RatingScreen.tsx',
  '/components/passenger/RechargeModal.tsx',
  '/components/passenger/RideCompletedScreen.tsx',
  '/components/passenger/RideHistoryScreen.tsx',
  '/components/passenger/RideInProgressScreen.tsx',
  '/components/passenger/RideScreen.tsx',
  '/components/passenger/RideTrackingScreen.tsx',
  '/components/passenger/SettingsScreen.tsx',
  '/components/passenger/SupportScreen.tsx',
  '/components/passenger/WalletScreen.tsx',
  '/components/passenger/DriverFoundScreen.tsx',
  '/components/passenger/LiveTrackingMap.tsx',
  '/components/passenger/DriverApproachingScreen.tsx',
  
  // Shared
  '/components/shared/UnifiedPolicyModal.tsx',
  
  // Pages
  '/pages/LandingPage.tsx',
  
  // Root
  '/1_NavigationScreen.tsx',
  '/2_EarningsScreen.tsx'
];

let fixed = 0;

files.forEach(filePath => {
  const fullPath = path.join(__dirname, filePath);
  
  try {
    if (!fs.existsSync(fullPath)) {
      console.log(`⚠️  Non trouvé: ${filePath}`);
      return;
    }
    
    let content = fs.readFileSync(fullPath, 'utf8');
    const original = content;
    
    // Déterminer le chemin relatif basé sur la profondeur
    const depth = filePath.split('/').length - 2; // -2 car on enlève le premier / et le nom du fichier
    const prefix = depth === 0 ? '.' : '../'.repeat(depth).slice(0, -1);
    
    // Corrections
    content = content.replace(/from ['"]framer-motion['"]/g, `from '${prefix}/framer-motion'`);
    content = content.replace(/from ['"]motion\/react['"]/g, `from '${prefix}/framer-motion'`);
    content = content.replace(/from ['"]framer-motion@[^'"]*['"]/g, `from '${prefix}/framer-motion'`);
    content = content.replace(/from ['"]lucide-react['"]/g, `from '${prefix}/lucide-react'`);
    content = content.replace(/from ['"]lucide-react@[^'"]*['"]/g, `from '${prefix}/lucide-react'`);
    
    if (content !== original) {
      fs.writeFileSync(fullPath, content, 'utf8');
      console.log(`✅ ${filePath}`);
      fixed++;
    }
  } catch (error) {
    console.log(`❌ Erreur sur ${filePath}:`, error.message);
  }
});

console.log('\n' + '='.repeat(60));
console.log(`✨ RÉSULTAT: ${fixed}/${files.length} fichiers corrigés`);
console.log('='.repeat(60) + '\n');

if (fixed > 0) {
  console.log('✅ Corrections appliquées avec succès!');
  console.log('🎨 Testez maintenant l\'application dans Figma Make\n');
} else {
  console.log('✅ Aucune correction nécessaire!\n');
}
